pkgs/build-support/singularity-tools/default.nix

```diff
@@ -20,40 +24,44 @@ rec {
       ${text}
     '';
 
-  mkLayer = {
-    name,
-    contents ? [],
-  }:
-    runCommand "singularity-layer-${name}" {
-      inherit contents;
-    } ''
+  mkLayer =
+    { name
+    , contents ? [ ]
+    ,
+    }:
+    runCommand "${projectName}-layer-${name}"
+      {
+        inherit contents;
+      } ''
       mkdir $out
       for f in $contents ; do
         cp -ra $f $out/
       done
     '';
 
-  buildImage = {
-    name,
-    contents ? [],
-    diskSize ? 1024,
-    runScript ? "#!${stdenv.shell}\nexec /bin/sh",
-    runAsRoot ? null,
-    memSize ? 512
-  }:
-    let layer = mkLayer {
-          inherit name;
-          contents = contents ++ [ bash runScriptFile ];
-          };
-        runAsRootFile = shellScript "run-as-root.sh" runAsRoot;
-        runScriptFile = shellScript "run-script.sh" runScript;
-        result = vmTools.runInLinuxVM (
-          runCommand "singularity-image-${name}.img" {
+  buildImage =
+    { name
+    , contents ? [ ]
+    , diskSize ? 1024
+    , runScript ? "#!${stdenv.shell}\nexec /bin/sh"
+    , runAsRoot ? null
```

Comment:
That sounds like a boolean. Maybe we should default it to false?

Reply:
This argument receives null-or-lines, and is meant to run with (pseudo-) root privileges.

It's created by @jbedo years ago. Removing these options may cause backward incompatibility.

Reply:
I wasn't suggesting to remove but to improve it. If it runs commands then it should be named something like ``runCommandsAsRoot`` and default to empty string if not set.

Reply:
I'll make another PR about `singularity-tools` after this one. Here I only format the Nix expression and replace `singularity` with `${projectName}`.